### PR TITLE
Get rid of FullTrust for ContextMenu and AppLaunch handlers

### PR DESCRIPTION
**Pull request by hez2010**
Let's do it in a progressive way.

As a beginning, aiming at ContextMenu and AppLaunch.

Opened an issue for traking: #9858

### CODE REVIEW

**Comment by yaichenbaum** [comment]
@hez2010 this is great! Have you noticed any performance improvements?

**Comment by hez2010** [comment]
> Have you noticed any performance improvements?

The loading speed of context menu is faster, though not that obvious. Also it doesn't require async loading, so the context menu will be fully loaded before showing.

**Comment by yaichenbaum** [comment]
> so the context menu will be fully loaded before showing.

I'll have to try this out but the current behavior of dispalying the menu before it finished loading was done on purpose since it there was often a two or three second delay.

**Comment by hez2010** [comment]
> there was often a two or three second delay.

I also noticed it. I'm making changes to make the context menu be loaded in the background.

**Comment by hez2010** [comment]
> I'm making changes to make the context menu be loaded in the background.

Done.

**Review by yaichenbaum — DISMISSED** [review]
The performance improvements here are amazing!

**Comment by hez2010** [comment]
Added a commit to cancel context menu loading eagerly to avoid duplicated entries in the context menu being added if user quickly right-clicks twice.

**Review by yaichenbaum — APPROVED** [review]

File: src/Files.App/Files.App.csproj [hunk 1]
```diff
@@ -23,6 +23,7 @@
     <NoWin32Manifest>True</NoWin32Manifest>
     <Platforms>x86;x64;arm64</Platforms>
     <RuntimeIdentifiers>win-x86;win-x64;win-arm64</RuntimeIdentifiers>
+    <LangVersion>10</LangVersion>
```
Comment (lukeblevins) [inline src/Files.App/Files.App.csproj:26]: Nit: Is there really a need to include this explicitly?

File: src/Files.App/BaseLayout.cs [hunk 2]
```diff
@@ -606,11 +606,10 @@ public async void BaseContextFlyout_Opening(object sender, object e)
                 if (!InstanceViewModel.IsPageTypeSearchResults && !InstanceViewModel.IsPageTypeZipFolder)
                 {
                     var shellMenuItems = await ContextFlyoutItemHelper.GetBaseContextShellCommandsAsync(currentInstanceViewModel: InstanceViewModel, workingDir: ParentShellPageInstance.FilesystemViewModel.WorkingDirectory, shiftPressed: shiftPressed, showOpenMenu: false, shellContextMenuItemCancellationToken.Token);
-                    if (shellContextMenuItemCancellationToken.IsCancellationRequested)
```
Comment (hez2010) [inline src/Files.App/BaseLayout.cs:609]: The old logic for detecting cancellation is invalid because it is always checking the new instanitated cts.

**Comment by hez2010** [comment]
I think this PR is ready for merge.
@yaichenbaum @lukeblevins PTAL.

**Review by yaichenbaum — APPROVED** [review]

**Comment by gave92** [comment]
It seems it does not show all the menu items, e.g with a zip file 7zip menu is not shown.
image

**Pull request merged**